Readme.md

```diff
@@ -12,61 +14,66 @@
 
-The general approach to this code has been to avoid premature performance optimization in favour of more maintainable code. When possible, it is also attempted to keep as much code outside of React and in regular JavaScript as possible. The idea is to make the code more approachable for non-React devs (Dustin, Kim), and to make more of the code reusable in future cases of framework swapouts.
+The general approach to this code has been to avoid premature performance optimization in favour of more maintainable code. When possible, attempts are made to keep as much code outside of React and in regular JavaScript as possible. The idea is to make the code more approachable for non-React devs, and to make more of the code reusable in future cases of framework swapouts or upgrades.
 
 #### Syncing data with the server
 
-Initialize navigation and app-load-based auto sync with the `useInitializeSyncApiDataIntoOfflineStorage` hook, get sync status state from the `useSyncStatus()` hook. Both hooks depend on `SyncStatusProvider`
-These auto syncs are triggered when navigating to the projects list page, and any time a user navigates to a project-related page. Note that not all syncs pull the same things. - the hooks take care of nagivation and reload based syncs.
+Initialize navigation and auto sync with the `useInitializeSyncApiDataIntoOfflineStorage` hook, get sync status state from the `useSyncStatus()` hook. Both hooks depend on `SyncStatusProvider`
+These auto syncs are triggered when navigating to the projects list page, and any time a user navigates to a project-related page. Note that not all syncs pull the same things. These hooks take care of nagivation and reload based syncs.
 
-There is also an `apiSyncInstance` available in the app for manualy controled syncs. Typically, in the databaseSwitchboard, creating, editing, or deleting Mermaid data will involve pushing the changes to IDB first, then pushing to the API, and then using the `pushThenPullAllProjectDataExceptChoices` to make sure a sync happens with all the data (beyond what was just pushed). The Project card component is an exception. There, the component code triggers a sync when the offline-ready checkbox is toggled. The component code handling sync, IDB, or API operations should generally be avoided. Whenever triggering a manual sync, make sure to set the sync status appropriately via `setIsSyncInProgress`
+There is also an `apiSyncInstance` available in the app for manualy controled syncs. Typically, in the databaseSwitchboard, creating, editing, or deleting Mermaid data will involve pushing the changes to IDB first, then pushing to the API, and then using the `pushThenPullAllProjectDataExceptChoices` to make sure a sync happens with all the data (beyond what was just pushed). Component code handling sync, IDB, or API operations should generally be avoided. Whenever triggering a manual sync, make sure to set the sync status appropriately via `setIsSyncInProgress`
```

Comment:
@saanobhaai what kind of wording would you use for 'choices'. Is it just literally API supplied input options/values? Or is there more to it?

I think we excluded it for many calls early on because the API was slow to generate it and it never/rarely changed. @gridcell ?

Reply:
^^ yup, data definitions would be very helpful

Reply:
Is this blocking this readme still, or do you agree that is for the back end docs?

Reply:
I don't think we need to deal with this in this readme; the code comments look ok for now. Also I don't think the choices endpoint is "slow"; but it is true its content rarely changes. So that's why the choices only get loaded once, when the app first loads (right?).